apps/common/models.py

```diff
@@ -44,3 +44,22 @@ class Meta(TypedModelMeta):  # type: ignore[reportIncompatibleVariableOverride]
     @typing.override
     def __str__(self):
         return str(self.pk)
+
+
+class ArchivableResource(Model):
```

Comment:
We also need to create a re-usable serializer for ArchivableResource. Similar to UserResource

We can then replace the logic for populating archived_at and archived_by for all the resources that are archivable.
The logic is already implemented for ContributorUserGroup